Wave1_data/1.preprocess_data/scripts/0.preprocess_raw_data.py

```diff
@@ -0,0 +1,580 @@
+    "0343": "O07",
+    "0344": "O08",
+    "0345": "O09",
+    "0346": "O10",
+    "0347": "O11",
+    "0348": "O12",
+    "0349": "O13",
+    "0350": "O14",
+    "0351": "O15",
+    "0352": "O16",
+    "0353": "O17",
+    "0354": "O18",
+    "0355": "O19",
+    "0356": "O20",
+    "0357": "O21",
+    "0358": "O22",
+    "0359": "O23",
+    "0360": "O24",
+    "0361": "P01",
+    "0362": "P02",
+    "0363": "P03",
+    "0364": "P04",
+    "0365": "P05",
+    "0366": "P06",
+    "0367": "P07",
+    "0368": "P08",
+    "0369": "P09",
+    "0370": "P10",
+    "0371": "P11",
+    "0372": "P12",
+    "0373": "P13",
+    "0374": "P14",
+    "0375": "P15",
+    "0376": "P16",
+    "0377": "P17",
+    "0378": "P18",
+    "0379": "P19",
+    "0380": "P20",
+    "0381": "P21",
+    "0382": "P22",
+    "0383": "P23",
+    "0384": "P24",
+}
+# write the well map to a json file
+path_to_repo_data = pathlib.Path("../../../data/processed/").resolve()
+path_to_repo_data.mkdir(exist_ok=True, parents=True)
+with open(path_to_repo_data / "well_map.json", "w") as f:
+    json.dump(well_map, f)
+# map the well to the well_map
+df["Well"] = df["Well"].map(well_map)
+df.head()
+
+
+# In[7]:
+
+
+print(f"There are {len(df['Well'].unique())} wells.")
+print(f"There are {len(df['FOV'].unique())} fields of view.")
+print(f"There are {len(df['Plate'].unique())} plates.")
+print(f"There are {len(df['Date_Time'].unique())} unique time points.")
+print("The times are:\n")
+print(df["Date_Time"].unique())
+
+
+# In[8]:
+
+
+# check that there are
+# 5 fovs * 5 channels * 96 wells = 2400 images per plate
+# get the dirs in the data directory
+dirs = glob.glob(str(pathlib.Path("../../../data/raw") / "*"))
+dirs = [x for x in dirs if pathlib.Path(x).is_dir()]
+plate_dict = {
+    "plate_name": [],
+    "num_files": [],
+}
+for dir in dirs:
+    # get the files in the dir
+    files = glob.glob(str(pathlib.Path(dir) / "*"))
+    files = [x for x in files if pathlib.Path(x).is_file()]
+    plate_dict["plate_name"].append(pathlib.Path(dir).name)
+    plate_dict["num_files"].append(len(files))
+plate_df = pd.DataFrame(plate_dict)
+plate_df["correct_num_files"] = plate_df["num_files"] == 2400
+# sort by correct_num_files
+plate_df.sort_values(by="correct_num_files", inplace=True)
+plate_df.reset_index(drop=True, inplace=True)
+plate_df
+
+
+# In[9]:
+
+
+# get all files in the 20241026T164425_ dir
+files = glob.glob(str(pathlib.Path("../../../data/raw/20241026T164425_") / "*"))
+files = [pathlib.Path(x).stem for x in files if pathlib.Path(x).is_file()]
+df = pd.DataFrame(files, columns=["file_name"])
+df["well"] = df["file_name"].str.split("F").str[0]
+df["FOV"] = df["file_name"].str.split("F").str[1].str.split("T").str[0]
+df["channel"] = df["file_name"].str.split("F").str[1].str.split("Z001").str[1]
+# sort by well, FOV, channel
+df.sort_values(by=["well", "FOV", "channel"], inplace=True)
+
+# get the value counts for well, FOV
+df[["well", "FOV"]].value_counts().sort_values()
```

Comment:
Recommend code comment for this section on why you are making these dfs for this specific directory and what you are looking to find. This applies to the code below for the other directory.

Reply:
Added some documentation in the form of a markdown cell